sdk/eventhub/event-hubs/test/partitionPump.spec.ts

```diff
@@ -31,6 +32,23 @@ describe("PartitionPump", () => {
         this.spanOptions = optionsArg;
         return super.startSpan(nameArg, optionsArg);
       }
+
+      withSpan<T extends (...args: unknown[]) => ReturnType<T>>(span: Span, fn: T): ReturnType<T> {
+        this.spanStack.push(span);
+        const result = fn();
+        if (typeof (result as any).then === "function") {
+          (result as any)
+            .then(() => this.spanStack.pop())
```

Comment:
I think you can just remove the .catch, right? Unless you think you're going to throw an error in 'pop'.